client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -483,6 +484,13 @@ const PluginSingleListView = ( {
 	const categories = useCategories();
 	const categoryName = categories[ category ]?.name || translate( 'Plugins' );
 
+	const installedPlugins = useSelector( ( state ) =>
+		getPlugins( state, siteObjectsToSiteIds( sites ) )
+	);
+	const aggregatedInstalledPlugins = useSelector( ( state ) =>
+		getPluginsOnSites( state, installedPlugins )
```

Comment:
Since `installedPlugins` already has the installed plugins in that site, what's the reason we need `aggregatedInstalledPlugins`?

Reply:
The idea sounds good but `aggregatedInstalledPlugins` is very expensive by itself. See `getPluginsOnSites` https://github.com/Automattic/wp-calypso/blob/8458f5119c43cfb194f994352d71654851d4aced/client/state/plugins/installed/selectors.js#L96 which includes `getPluginOnSites` which includes `.find()`! 

In my tests it hovers around 4-5ms. Whereas by using another version of `isNotInstalled` I can get timings < 0.1ms.

```
const t0 = performance.now();
plugins = plugins
	.filter( isNotBlocked )
	.filter( ( plugin ) => isNotInstalled( plugin, installedPlugins ) );
const t1 = performance.now();
console.log( `Call to isNotInstalled took ${ t1 - t0 } milliseconds.` );

function isNotInstalled( plugin, installedPlugins ) {
	return ! installedPlugins.find( ( item ) => item.slug === plugin.slug );
}
```

I guess that if we were to check thousand or millions of plugins, aggregating them would make sense.

Reply:
Thanks for gathering this profile information. I applied this suggestion here https://github.com/Automattic/wp-calypso/pull/63712/commits/c67000376f1b15f8ce4145dfaf4452e0edff3a40